### PR TITLE
fix(quantic): Search Interface not triggering an empty search request when pressing browser back button

### PR DESCRIPTION
**Pull request by SimonMilord**
[SFINT-5539](https://coveord.atlassian.net/browse/SFINT-5539)

### ISSUE:
- When navigating to the Hosted search page from the hosted searchbox with an empty query, it would push `this.urlManager.state.fragment  => ''` in the window history which would mess with the redirection when pressing the browser back button since its not `#q=`. It would redirect from HSP with `#` in url to HSP with `#q=` in the url.

BEFORE:

https://github.com/coveo/ui-kit/assets/73316533/fde48eb1-3826-4e6d-b0e6-02526484ac28

### SOLUTION:
- Modified the `updateHash` method to check if `this.urlManager.state.fragment === ''` before pushing the fragment from the url Manager to the window history. If its equal to '' then it returns instead of adding it to the window history.
- Will add E2E test for this edge case in the SFINT package in the HSP/Hosted searchbox test suite.

### DEMO:

https://github.com/coveo/ui-kit/assets/73316533/7888b1ac-2f79-4833-a843-e1259e28836e

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticSearchInterface/quanticSearchInterface.js [hunk 1]
```diff
@@ -181,6 +181,9 @@ export default class QuanticSearchInterface extends LightningElement {
   }
 
   updateHash() {
+    if (this.urlManager.state.fragment === '') {
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (SimonMilord) [reply packages/quantic/force-app/main/default/lwc/quanticSearchInterface/quanticSearchInterface.js:186]: hmmm ok will continue to dig

[the remark this answers is not in the document]
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticSearchInterface/quanticSearchInterface.js:186]: Thank you! I can help when I will have time.
